brave-core/src/main/java/com/github/kristofa/brave/internal/Propagation.java

```diff
@@ -0,0 +1,51 @@
+package com.github.kristofa.brave.internal;
+
+import com.github.kristofa.brave.SpanId;
+import com.github.kristofa.brave.TraceData;
+
+/**
+ * Injects and extracts {@link SpanId span identifiers} sent in-band across process boundaries.
+ *
+ * <p>Implementations usually wrap Http, Messaging or RPC headers.
+ *
+ * @param <K> Usually, but not always a String
+ */
+public interface Propagation<K> {
```

Comment:
this type and methods prefixed internal will stay that way until the out-of-band pull request #276 is done.